app/models/trace.rb

```diff
@@ -233,13 +233,13 @@ def import
           tracepoints << tp
         end
 
-        # Run the before_save and before_create callbacks, and then import them in bulk with activerecord-import
+        # Run the before_save and before_create callbacks, and then do a bulk insert
         tracepoints.each do |tp|
           tp.run_callbacks(:save) { false }
           tp.run_callbacks(:create) { false }
         end
 
-        Tracepoint.import!(tracepoints)
+        Tracepoint.insert_all!(tracepoints.map(&:attributes))
```

Comment:
I've done some testing and I think this looks right - my main concern is the timestamp column as `attributes` returns it in string form.

It would probably be a good idea to add a test to the trace model tests that imports `a.gpx` and checks that the one point it imports has the expected values for each attribute as we don't seem to have one that checks the imported data at the moment.

Reply:
Good idea, I've added some unit tests for the API endpoint. It's a bit unfortunate though, that lat/lon values are identical in  `a.gpx`.

I've also added a call to `validate!`. I'm not sure if this whole validation is working as expected. Like before, I'm getting import failures in case the timestamp is missing, which is good. However, missing lat/lon values are not reported neither in the current code nor in the updated version in this PR.

Reply:
There are no validation on lat/lon at the moment so that's not surprising...

Reply:
Ok, good. I believe all open points should be addressed by now.